### PR TITLE
Updating form block

### PR DESCRIPTION
**Pull request by jalagari**
Using latest form block describe in http://main--afb--adobe.hlx.live/ for creating Form Edge Delivery form story.

Test URLs:
https://main--wknd--hlxsites.hlx.page/drafts/jalagari/demo/home

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -1,8 +1,250 @@
-import { analyticsTrackFormSubmission } from '../../scripts/analytics/lib-analytics.js';
+/**
+ * copy it from lib-franklin to avoid dependency. // TODO - Do not mergin into main branch
+ */
+export function toClassName(name) {
```
Comment (dylandepass) [inline blocks/form/form.js:4]: Should probably import from `lib-franklin` now.

File: blocks/form/form.js [hunk 2]
```diff
@@ -1,8 +1,250 @@
...
+  return typeof name === 'string'
+    ? name.toLowerCase().replace(/[^0-9a-z]/gi, '-').replace(/-+/g, '-').replace(/^-|-$/g, '')
+    : '';
+}
+
+/**
+ *  copy it from lib-franklin to avoid dependency. // TODO - Do not mergin into main branch
+ */
+export function readBlockConfig(block) {
```
Comment (dylandepass) [inline blocks/form/form.js:13]: Same here

**Review by dylandepass — CHANGES_REQUESTED** [review]
@jalagari I'm not familiar with what `adobeaem.workers.dev` is doing.. but I just want to confirm that with this PR we are not removing the ability to submit the form directly to the pipeline (and not through this forms worker).

Assuming so, I think we should have an example of both. That way sales folks don't need to know anything about `franklin-submit-wrapper.adobeaem.workers.dev` and can submit forms without it for easy demo purposes.

**Comment by jalagari** [comment]
> @jalagari I'm not familiar with what `adobeaem.workers.dev` is doing.. but I just want to confirm that with this PR we are not removing the ability to submit the form directly to the pipeline (and not through this forms worker).
> 
> Assuming so, I think we should have an example of both. That way sales folks don't need to know anything about `franklin-submit-wrapper.adobeaem.workers.dev` and can submit forms without it for easy demo purposes.

@dylandepass - By default, Submit button will always invoke Form Service (pipeline) but author can override the submit endpoint at time of defining Form block in word document ( like any other block configuration). In this specific example we are call franklin-form-wrapper which upload the attachments ( if any) and forward call to form service (pipeline).

**Comment by dylandepass** [comment]
> @dylandepass - By default, Submit button will always invoke Form Service (pipeline) but author can override the submit endpoint at time of defining Form block in word document ( like any other block configuration). In this specific example we are call franklin-form-wrapper which upload the attachments ( if any) and forward call to form service (pipeline).

Perfect, wanted to verify the POST payload would still support a submit to the pipeline. There is a lot of code there so it wasn't immediately obvious that was the case.

**Review by dylandepass — APPROVED** [review]

**Pull request merged**